plugins/catalog-backend-module-example-processor/src/processor/ExampleProcessor.ts

```diff
@@ -0,0 +1,38 @@
+import { ComponentEntity, Entity } from '@backstage/catalog-model';
+
+import { LocationSpec } from '@backstage/plugin-catalog-common';
+import { CatalogProcessor } from '@backstage/plugin-catalog-node';
+
+export class ExampleProcessor implements CatalogProcessor {
+  getProcessorName(): string {
+    return 'ExampleProcessor';
+  }
+
+  async preProcessEntity(
+    entity: Entity,
+    location: LocationSpec,
+  ): Promise<Entity> {
+    if (location.type !== 'url') {
+      return entity;
+    }
+
+    if (entity.metadata.name === 'backstage') {
+      return entity;
+    }
+
+    const managedByLocation =
+      entity.metadata.annotations?.['backstage.io/managed-by-origin-location'];
+    if (
+      managedByLocation &&
+      (managedByLocation ===
+        'url:https://github.com/backstage/community-plugins/blob/main/**/catalog-info.yaml' ||
+        managedByLocation ===
+          'url:https://github.com/backstage/backstage/blob/master/**/catalog-info.yaml')
```

Comment:
these matchers are a bit fragile (if we change the manner of registering them or change the default branch name or so). Is it worth making these just match by prefix (eg the `url:https://github.com/backstage/backstage/` bit)?

Reply:
I had done this to avoid hitting any of the current examples from: https://github.com/backstage/backstage/tree/master/packages/catalog-model/examples. For now let's leave it but you are correct, it's not great.